### PR TITLE
[reflect] Fix wrong argument size when calling default instance methods.

### PR DESCRIPTION
**Pull request by XYZboom**
KT-71378 fixed.
Since there are other failed tests here that can now pass, no more tests have been created.

### CODE REVIEW

File: core/reflection.jvm/src/kotlin/reflect/jvm/internal/calls/ValueClassAwareCaller.kt [hunk 1]
```diff
@@ -112,12 +116,13 @@ internal class ValueClassAwareCaller<out M : Member?>(
 
         fun typeSize(type: KotlinType): Int = getMfvcUnboxMethods(type.asSimpleType())?.size ?: 1
```
Comment (udalov) [inline core/reflection.jvm/src/kotlin/reflect/jvm/internal/calls/ValueClassAwareCaller.kt:117]: Please inline this function

File: core/reflection.jvm/src/kotlin/reflect/jvm/internal/KFunctionImpl.kt [hunk 2]
```diff
@@ -168,6 +173,11 @@ internal class KFunctionImpl private constructor(
     private val boundReceiver
         get() = rawBoundReceiver.coerceToExpectedReceiverType(descriptor)
 
+    private fun createDefaultInstanceBoundStaticCaller(member: Method) =
```
Comment (udalov) [inline core/reflection.jvm/src/kotlin/reflect/jvm/internal/KFunctionImpl.kt:176]: `createBoundStaticCaller`

Comment (udalov) [reply core/reflection.jvm/src/kotlin/reflect/jvm/internal/KFunctionImpl.kt:176]: Also, please add explicit return type

File: core/reflection.jvm/src/kotlin/reflect/jvm/internal/calls/CallerImpl.kt [hunk 3]
```diff
@@ -131,6 +131,12 @@ internal sealed class CallerImpl<out M : Member>(
             }
         }
 
+        /**
+         * A default caller whose original caller is a instance method.
+         * see [KT-71378](https://youtrack.jetbrains.com/issue/KT-71378)
```
Comment (udalov) [inline core/reflection.jvm/src/kotlin/reflect/jvm/internal/calls/CallerImpl.kt:136]: I don't understand this comment. What's "original caller" and why is it an instance method? `copy` is a static method.

**Review by udalov — CHANGES_REQUESTED** [review]
^

File: core/reflection.jvm/src/kotlin/reflect/jvm/internal/calls/ValueClassAwareCaller.kt (hunk 1 above)
Comment (XYZboom) [reply core/reflection.jvm/src/kotlin/reflect/jvm/internal/calls/ValueClassAwareCaller.kt]: done

File: core/reflection.jvm/src/kotlin/reflect/jvm/internal/calls/CallerImpl.kt (hunk 3 above)
Comment (XYZboom) [reply core/reflection.jvm/src/kotlin/reflect/jvm/internal/calls/CallerImpl.kt]: `copy(Duration, Bool)` (`copy(long, boolean)` on jvm) is an instance method and the generated method `copy$default(Test, long, boolean, int, Object)` is static. 
We are creating default caller which calls `copy$default` so **original caller** here is the instance method `copy`. 
Necessary comments have been added

File: core/reflection.jvm/src/kotlin/reflect/jvm/internal/KFunctionImpl.kt (hunk 2 above)
Comment (XYZboom) [reply core/reflection.jvm/src/kotlin/reflect/jvm/internal/KFunctionImpl.kt]: done

**Review by udalov — APPROVED** [review]
I've reworked the change a bit

**Review by XYZboom — COMMENTED** [review]
Thanks for the improvements!

**Comment by udalov** [comment]
Merged with a few changes in https://github.com/JetBrains/kotlin/commit/ec18ba7fccf5022f067ac81343839be90a5b7edf, thank you!

**Pull request closed without merging**